build/test-integration.sh

```diff
@@ -36,7 +36,9 @@ integrationTestPlatform=""
 
 if [ -n "$1" ]; then
     testCaseFilter="--filter $1"
-    integrationTestPlatform="."$(echo $1 | cut -d'=' -f 2)
+	if [ -n "$BUILD_NUMBER" ]; then
```

Comment:
this check by itself doesn't make sense to me. Are you using BUILD_NUMBER as a proxy for "running on an agent as opposed to dev box"? If yes, let's add a variable in __variables.sh to express this directly, and use that variable instead.

Reply:
We currently use this in our other scripts. (e.g. build scipts for build and runtime images)
Curious: what is the advantage of doing that?  We can change other scripts also then